### PR TITLE
CsvEnumerable

### PR DESCRIPTION
**Pull request by EugeniaLos**

### CODE REVIEW

File: DotnetFundamentals/CsvEnumerable/CsvEnumerable.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+        private class CsvEnumarator: IEnumerator
+        {
+            private string[] records;
+            private int position = -1;
+
+            public CsvEnumarator(List<string> records)
+            {
+                this.records = new string[records.Count];
+                records.CopyTo(this.records);
```
Comment (Happi-cat) [inline DotnetFundamentals/CsvEnumerable/CsvEnumerable.cs]: а почему так?

File: DotnetFundamentals/CsvEnumerable/CsvEnumerable.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            return this.GetEnumerator();
+        }
+
+
+
+        private class CsvEnumerator<T>: IEnumerator<T>
+        {
+            private bool firstRecord;
+            private string fileName;
+            private StreamReader sr;
```
Comment (SergeyGontsarevichITechart) [inline DotnetFundamentals/CsvEnumerable/CsvEnumerable.cs]: Please use the same name format for all fields.

File: DotnetFundamentals/CsvEnumerable/CsvEnumerable.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            {
+                this.fileName = fileName;
+                firstRecord = true;
+            }
+
+            public bool MoveNext()
+            {
+                if (firstRecord)
+                {
+                    sr = new StreamReader(Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName, fileName));
```
Comment (SergeyGontsarevichITechart) [inline DotnetFundamentals/CsvEnumerable/CsvEnumerable.cs]: I want to suggest move logic from CsvEnumerable that combines path for file name.

File: DotnetFundamentals/CsvEnumerable/CsvEnumerable.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+            object IEnumerator.Current
+            {
+                get
+                {
+                    return csvReader.GetRecord<T>();
+                }
+            }
+
+            public void Reset()
```
Comment (SergeyGontsarevichITechart) [inline DotnetFundamentals/CsvEnumerable/CsvEnumerable.cs]: Please add reset for firstRecord field

File: DotnetFundamentals/CsvEnumerable/CsvEnumerable.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return this.GetEnumerator();
+        }
+
+
+
+        private class CsvEnumerator<T>: IEnumerator<T>
```
Comment (SergeyGontsarevichITechart) [inline DotnetFundamentals/CsvEnumerable/CsvEnumerable.cs]: Please merge CsvEnumerator and CsvEnumerable to the one class

File: DotnetFundamentals/CsvEnumerable/CsvEnumerable.cs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+        public bool MoveNext()
+        {
+            if (firstRecord)
+            {
+                streamReader = new StreamReader(filePath);
+                csvReader = new CsvReader(streamReader);
+                firstRecord = false;
+            }
+        return csvReader.Read();
```
Comment (Happi-cat) [inline DotnetFundamentals/CsvEnumerable/CsvEnumerable.cs]: ident

**Pull request merged**